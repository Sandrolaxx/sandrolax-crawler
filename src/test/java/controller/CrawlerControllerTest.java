package controller;

import javax.ws.rs.core.Response.Status;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.config.HttpClientConfig;
import io.restassured.http.ContentType;
import io.restassured.specification.RequestSpecification;

/**
 *
 * @author SRamos
 */
@QuarkusTest
public class CrawlerControllerTest {

    private RequestSpecification given() {
        var config = RestAssured.config()
                .httpClient(HttpClientConfig.httpClientConfig()
                        .setParam("http.connection.timeout", 90000)
                        .setParam("http.socket.timeout", 90000));

        return RestAssured.given().config(config).contentType(ContentType.JSON);
    }

    @Test
    public void whenGetCrawler() {
        var result = given()
                .headers("review_star", "1")
                .when().get("/crawler")
                .then()
                .statusCode(Status.OK.getStatusCode())
                .extract().asString();

        Approvals.verify(result);
    }

}
