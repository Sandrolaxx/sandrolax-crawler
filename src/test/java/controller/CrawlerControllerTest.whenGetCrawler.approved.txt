[{"title":"Battlefield Earth","releaseDate":"2000","rating":2.5,"direction":[{"director":"Roger Christian"}],"cast":[{"actor":"John Travolta"},{"actor":"Forest Whitaker"},{"actor":"Barry Pepper"},{"actor":"Kim Coates"},{"actor":"Sabine Karsenti"},{"actor":"Michael Byrne"},{"actor":"Christian Tessier"},{"actor":"Sylvain Landry"},{"actor":"Richard Tyson"},{"actor":"Christopher Freeman"},{"actor":"John Topor"},{"actor":"Shaun Austin-Olsen"},{"actor":"Tim Post"},{"actor":"Earl Pastko"},{"actor":"Michel Perron"},{"actor":"Michael MacRae"},{"actor":"Todd McDougall"},{"actor":"Derrick Damon Reeve"}],"comment":{"title":"One of the greatest sci fi comedies ever made!","review":"My spouse and I went to see this on opening night. We were expecting to see an extremely bad and costly film, and we were not really disappointed. It is a testament to John Travolta's genius that his career survived this disaster at all. As they say, garbage-in, garbage out. Start with an L Ron Hubbard novel (your first mistake) featuring a completely plagiarized dark-skinned, war-like and hairy alien culture with wrinkly foreheads (if this sounds like Klingons to you, I thought so too!), and add unfortunate Barry \"can't get a break\" (or maybe it should be 'cant pick a script') Pepper and all you need is Travolta himself - playing the arrogant, merciless, slightly effeminate and quite under-sized leader of the alien colonists. Remove any hint of character development from the script and use the worst of the worst black box technobabble explanations for plot devices. Finally, stay true to the idiotic gibberish you based the film on in the first place, and you've got the makings of a rolling-on-the-floor comedy. In fact, opening night, in a packed theater, people started laughing out loud about 20 minutes into the film and never really stopped. We had a great time that night. Halfway through the film more than half the crowd was actually interacting with the film, asking \"Mr Worf, where's Commander Riker?\" and asking troublesome questions about how many hundreds of years gas would remain viable in the gas-tank of a Harrier. If you need to know about the plot - it's this simple - Earth has been conquered by an amazingly stupid group of Klingon-like aliens, and the remaining humans live in a large domed slave labor camp where they are taught that their sole purpose is servitude because they are stupid, weak, etc. Barry Pepper somehow becomes convinced that he's not born to be a slave, and learns to fly a harrier, etc. It's really not worth the effort of typing. There are a few not-so-subtle and not very original but good messages about ethnocentrism to be found here, but not much else. Some day when you need a good laugh, rent this or borrow it from your local bad film collector."}},{"title":"Epic Movie","releaseDate":"2007","rating":2.4,"direction":[{"director":"Jason Friedberg"},{"director":"Aaron Seltzer"}],"cast":[{"actor":"Kal Penn"},{"actor":"Jennifer Coolidge"},{"actor":"Fred Willard"},{"actor":"Adam Campbell"},{"actor":"Jayma Mays"},{"actor":"Faune Chambers Watkins"},{"actor":"Crispin Glover"},{"actor":"Tony Cox"},{"actor":"Héctor Jiménez"},{"actor":"Darrell Hammond"},{"actor":"Carmen Electra"},{"actor":"David Carradine"},{"actor":"Kevin McDonald"},{"actor":"George Alvarez"},{"actor":"Crista Flanagan"},{"actor":"Dana Seltzer"},{"actor":"Dane Farwell"},{"actor":"Tad Hilgenbrink"}],"comment":{"title":"This thing won't let me vote below a 1....","review":"After watching this movie, I just couldn't stop talking about it. Mind you, not in the sense that someone can't stop talking about a great book or show, but more along the line of how someone can't stop talking about their colonoscopy during a diner conversation. In other words, after seeing this you can't help but try and remove the stain it left on your very soul by vomiting out the experience onto others. This movie entered with the tag line of being \"written by two of the six writers of 'Scary Movie'!\" What they forgot to mention is that the two who wrote it were only responsible for writing the credits and all the jokes deleted for the sake of not causing mass suicide. Put more delicately, this movie almost makes Pootie Tang look Oscar worthy, which ironically enough was made fun of in the last Scary Movie for being horrid. Coincidence? Yeah, not even some sort of Karma dealing fiend would wish this movie upon the world. \"So,\" you ask, \"What makes it so bad?\" I bet your thinking that all of the funny stuff was in the previews, right? No, actually there was no funny content in the previews at all, it just seemed that way when taken out of the context of the movie. Much the same way \"Hogan's Heroes\" was funny to people who weren't in Auschwitz during the Holocaust, the previews are funny to those who haven't seen the movie. When you actually see the movie, you start to ask, \"Why was that funny in the preview?\" The answer: You can see pretty flowers if you ignore the fact that they are growing from the corpse of a man who died during a freak green house accident. Do yourself a favor, rent a History channel documentary on the history of documentaries and hit yourself in the head with a hammer. It'll be twice as funny, make three times more sense, and be only a fifth as painful."}},{"title":"Son of the Mask","releaseDate":"2005","rating":2.2,"direction":[{"director":"Lawrence Guterman"}],"cast":[{"actor":"Jamie Kennedy"},{"actor":"Traylor Howard"},{"actor":"Alan Cumming"},{"actor":"Liam Falconer"},{"actor":"Ryan Falconer"},{"actor":"Steven Wright"},{"actor":"Kal Penn"},{"actor":"Ben Stein"},{"actor":"Brett Pickup"},{"actor":"Peter Callan"},{"actor":"Ashley Lyons"},{"actor":"Wayne McDaniel"},{"actor":"Sandy Winton"},{"actor":"Rebecca Massey"},{"actor":"Issac Longmuir"},{"actor":"Tayzin Fahey-Leigh"},{"actor":"Skyla Laginha"},{"actor":"Lochie Nazer Hennings"}],"comment":{"title":"Son of the Mask: O....M....G!","review":"Son of the Mask is one of those sequels that simply shouldn't have happened and now after watching I can claim it's a contender for the worst one of the bunch. But no wait there is more, this isn't just bad it's THAT bad. It currently sits as the 10th worst movie ever made on IMDB which is an impressive feat. Ontop of that it's the winner of such prestigious awards as Worst Picture, Worst Director, Worst Actor, Most Intrusive Musical Score, Foulest Family Film, Most Painfully Unfunny Comedy and Least \"Special\" Special Effects. Starring Jamie Kennedy, Traylor Howard, Alan Cumming and a small role for Kal Penn it tells another story of the infamous Mask and it is beyond belief. Full of dire cgi, painfully unfunny \"comedy\", purile writing and legitimately no redeeming features I can't believe that such a brilliant classic comedy like The Mask (1994) could be insulted to this degree with a sequel. It's just SO cringe inducing! The jokes are too immature even for Saturday morning cartoons. The cartoon violence doesn't translate very well at all and some of it seems distasteful especially for a family film. And for 84 million dollars it looks like warmed up dog poop. Seriously, everything people say about this film is true. I'm not one to jump aboard bandwagons and hate things because its \"Trendy\" to, nor do I give 1/10's out often but this is awful! The Good: Traylor Howard The Bad: Poor cgi Beyond dumb Things I Learnt From This Movie: In movies reducing women down to their bras is funny and comedic, I do it and I get arrested!? How could a plot including repeated attempts on a babies life not be considered funny? Shocking! Children being stolen by norse gods isn't even remotely shocking to some parents Odin is a WWE champion"}},{"title":"House of the Dead","releaseDate":"2003","rating":2.1,"direction":[{"director":"Uwe Boll"}],"cast":[{"actor":"Jonathan Cherry"},{"actor":"Tyron Leitso"},{"actor":"Clint Howard"},{"actor":"Ona Grauer"},{"actor":"Ellie Cornell"},{"actor":"Will Sanderson"},{"actor":"Enuka Okuma"},{"actor":"Kira Clavell"},{"actor":"Sonya Salomaa"},{"actor":"Michael Eklund"},{"actor":"David Palffy"},{"actor":"Jürgen Prochnow"},{"actor":"Steve Byers"},{"actor":"Erica Durance"},{"actor":"Birgit Stein"},{"actor":"Jay Brazeau"},{"actor":"Adam J. Harrington"},{"actor":"Colin Lawrence"}],"comment":{"title":"Lord have mercy!","review":"There are people out there who will greenlight anything! That is the only explanation I can offer as to why the House of the Dead movie exists. And that's only scary part to the whole movie. It's so bad you'll go off movies forever. I seriously wanted to switch this off and turn the TV over to the Paint Drying channel but I was bound by my word to suffer the whole thing. I don't know why I do these bad things to myself. As if it matters, here's the basic jist of the 'story'. A group of twenty-somethings are so desperate to go out to some island in the Pacific Northwest (Canada actually, because it's cheap) for the 'Rave of the Century' (which consists of about 8 people and un-raving music) that they pay some craggy old fisherman $1000 to take them there after they miss the main ferry. That's gotta be some rave to be worth all that dough! The fisherman warns them that the island is also known as the Island of the Dead (hang on-I thought this was HOUSE of the Dead?) and that they are all doomed yadda yadda yadda. First faults here. Why would a tiny little rave (of the Century my foot!) be held on some remote island? Why would anyone willingly pay loads of money to get it? Why pay even more to the craggy old fisherman to take them back when they could just come back with the others? Once they arrive they discover that the rave (which consists of about 2 tents, a small stage and a port-a-john) has been smashed, there's blood everywhere and no one is around. What would any rationally thinking person do? Run for their lives of course. But no, these clueless, obviously blind people decide to go look for them. Soon enough they discover an old ramshackle house that's 50 times as big on the inside as it is on the outside. Another half hour of stumbling around in the forest follows, as an excuse to kill of some of the lesser characters, and after much tedium they arrive back at the house again. The characters, like the movie, go nowhere. Jammed into this ghastly disaster is a superabundance of gibberish dialogue, heinous acting, mumbo-jumbo exposition and zillions of clips from the once-popular arcade game of the same name. Why this was universally accepted as a good idea with the filmmakers I'll never know. The clips have no reference to any of the scenes and only degrade this trash even further, if that is at all possible. It has nothing to do with the game save for some cheap, throwaway line at the end. It makes Resident Evil look like cinematic glory. Hell, even the Double Dragon movie seems multi-Oscar worthy in comparison to this junk. The only one who comes out of this with his dignity still intact is Jurgen Prochnow. He could have just taken his money and ran but he tries his best with the awful script and brings a tiny bit of pathos to his character. The rest of the cast suck I'm afraid. The characters are idiots and deserve to die. Plus, if you cut out the swearing and pointless nudity, I see no reason why this film cannot be shown on Saturday morning TV. It's not frightening in the slightest. Pirates of the Caribbean is more scary than the skeletal bad guys in this film. And where did all those bad guys come from anyway? There were only a few people on the island to begin with. I guess this justifies the reason they chose to reuse footage over and over. I kid you not, you'll see the same zombie die a dozen times. Who's ultimately to blame for that scandalous waste of celluloid? None other than director Uwe Boll. His control over the movie is non-existent. You can clearly the see actors have no idea what they should be doing and that the zombies aren't really taking it all seriously. The actors seem like they're reading off cue cards as they constantly pause in the middle of long sentences and carry on talking as soon as they see the next card. It all feels very unnatural. Plus the film is shot like a two-part mini-series. I have indeed seen better TV productions. And don't get me started on the editing. The film is an incoherent babble with thousands upon thousands of pointless shots and dozens of meaningless camera pans. No real skill or talent was put into making this at all. It truly baffles and boggles the mind how movies this unfathomably bad can get made and George A. Romero can't even get anyone to take his calls. House of the Dead makes some idiotic reference to Romero in a lazy attempt to be 'post-modern' but it only irritates that they think THIS is in the same league as a REAL zombie movie. For what it's worth, the 1.85:1 anamorphic picture looks great and the Dolby 5.1 soundtrack is clean but very unimpressive and only serves to pronounce the heavily over-used ADR even more. The DVD comes with extras but why torture yourself. Isn't this review warning enough? Stay away! You are all doomed I tell you! Doomed! Doomed!!!"}},{"title":"Disaster Movie","releaseDate":"2008","rating":1.9,"direction":[{"director":"Jason Friedberg"},{"director":"Aaron Seltzer"}],"cast":[{"actor":"Carmen Electra"},{"actor":"Vanessa Lachey"},{"actor":"Nicole Parker"},{"actor":"Matt Lanter"},{"actor":"Gary 'G. Thang' Johnson"},{"actor":"Crista Flanagan"},{"actor":"Kim Kardashian"},{"actor":"Ike Barinholtz"},{"actor":"Tony Cox"},{"actor":"Tad Hilgenbrink"},{"actor":"Nick Steele"},{"actor":"John Di Domenico"},{"actor":"Jason Boegh"},{"actor":"Valerie Wildman"},{"actor":"Abe Spigner"},{"actor":"Noah Harpster"},{"actor":"Austin Scott"},{"actor":"Devin Crittenden"}],"comment":{"title":"My Eyes!!!!","review":"It's hard to imagine anyone writing a 'spoof' movie even worse than Meet the Spartans, but Seltzer and Friedberg have pulled it off. It's not even just a little worse. It makes Spartans look like Raiders of the Lost Ark in comparison. You will seriously feel ill afterwards and more than likely have a strong urge to gouge your eyes out. May Contain Spoilers....if you care: First off, as it appeared in the trailer, this movie does not spoof, or really even reference disaster movies at all. The set shakes a few times, there are some plastic asteroids, and a bad cgi tornado, all of which that do nothing to add anything to the movie, but other than this there's nothing even relating this to said style of film. Instead we get references (not to be confused with actual spoofs) to such films as High School Musical, Step Up, this summer's crop of comic based films, Enchanted, Alvin and the Chipmunks, Kung Fu Panda, and Sex and the City. I can't wait to hear Seltzer and Friedberg praise their own genius and describe this movie as being modeled around Cloverfield...they clearly tried to do this, but other than the outline of the small group of people running through the city trying to find the injured girlfriend, there's nothing relating this piece of crap to the monster movie. I can honestly say I did not laugh once at this movie. All of the \"jokes\" were forced, and many of them were dragged on for several minutes at a time...the two writers clearly thinking they were onto some kind of comedy gold here. Probably the most insulting thing about this movie is, like all their other gems: Epic, Date, Spartans, they consider the audience to be mentally challenged...having to tell us point blank what movie it is they are referencing (again...not spoofing) Such lines like: \"It's an inconvenient truth, it's global warming,\" \"Come here you kung fu panda,\" \"what is this enchanted place,\" \"How do you know all this? I know because I saw Indiana Jones and the Kingdom of the Crystal Skull,\" \"Let me get my Get Smart phone (proceeds to talk on his shoe),\" and \"whats with the marching band? I thought I'd cheer you up with a high school musical\" all exist to make sure we're in on their super intelligent antics. Please, save your money for anything else. Show Hollywood that they need to stop these two guys from making any more movies. I can't even imagine a 5 year old finding this stuff funny. Speaking of 5 years old, that's about how long many of these jokes will last before absolutely no one can find them funny anymore since, as usual, most of them are simply based on pop culture references."}},{"title":"The Hottie & the Nottie","releaseDate":"2008","rating":1.9,"direction":[{"director":"Tom Putnam"}],"cast":[{"actor":"Paris Hilton"},{"actor":"Joel David Moore"},{"actor":"Christine Lakin"},{"actor":"Johann Urb"},{"actor":"Adam Kulbersh"},{"actor":"Greg Romero Wilson"},{"actor":"Marianne Muellerleile"},{"actor":"Kathryn Fiore"},{"actor":"Scott Prendergast"},{"actor":"Morgan Rusler"},{"actor":"Ryan Alvarez"},{"actor":"Erin Cardillo"},{"actor":"Samantha Bailey"},{"actor":"Jeremy Scott Johnson"},{"actor":"Gino Anthony Pesi"},{"actor":"Karley Scott Collins"},{"actor":"Caleb Guss"},{"actor":"Kurt Doss"}],"comment":{"title":"why would you do this to people?","review":"this movie is to say the least, one of the worst things done to mankind in the past 100 years right behind the a-bomb. I had the misfortune of seeing this piece of garbage with my sister for her birthday. I would not recommend seeing this movie even it it is for comic value of how bad it is. There are 3 scales of bad: 1. the kind that makes you laugh it is so bad (murdercycle) 2. the kind that is soo bad it pisses you off (Epic Movie) 3. Gary busey bad (The hottie and the nottie). i hated it but my sister loved it, (granted she is 12 and has down syndrome (no lie)). the fact that this movie also tries to portray a message is even worse, because i wasn't sure what it was until i did some reading.... this is perfect grounds why IMDb needs a 0/10"}},{"title":"Birdemic: Shock and Terror","releaseDate":"2010","rating":1.7,"direction":[{"director":"James Nguyen"}],"cast":[{"actor":"Alan Bagh"},{"actor":"Whitney Moore"},{"actor":"Tippi Hedren"},{"actor":"Janae Caster"},{"actor":"Colton Osborne"},{"actor":"Adam Sessa"},{"actor":"Catherine Batcha"},{"actor":"Patsy van Ettinger"},{"actor":"Damien Carter"},{"actor":"Rick Camp"},{"actor":"Stephen Gustavson"},{"actor":"Danny Webber"},{"actor":"Mona Lisa Moon"},{"actor":"Joe Teixeira"},{"actor":"John Grant"},{"actor":"Natalie Yonkers"},{"actor":"Laura Cassidy"},{"actor":"Bonnie Steiger"}],"comment":{"title":"A \"must-see\" movie. Really. I'm not kidding.","review":"Don't be fooled by the rating. You have to watch \"Birdemic\" at some point. This movies outshines \"Troll 2\" in every respect. It puts \"Plan 9 From Outer Space\" and \"The Room\" to shame. Imagine a movie that would be created solely to give film students a paper topic on how NOT to make movies. Every conceivable error has been made in making \"Birdemic.\" At first, you simply won't be able to believe that the film is not a spoof. But if you read about the film and about Nguyen, and if you take a few moments to watch the additional material on the DVD, you will soon agree that this was just a misguided effort on the entire production team's part. The sound is the most outstanding example of sheer incompetence. The director clearly understood what Foley is, since the gunshots have all been added post-production, but I guess they just didn't have the time or money to dub the vast majority of the film after shooting. And it needs it. Badly. The acting is uniformly terrible. Not bad. Terrible. Alan Bragh doesn't even rise to the level of fourth grade pageants. Whitney Moore is cute as a button and is aided by a bit of camel-toe in at least one scene, but she's the best of a poor lot. All the supporting cast is \"give your financial backers a small part\" bad. If you want to explain blocking, editing, framing, sound markers, continuity, reverse shots, and other film terms to your friends, this is the film to use. None of it is done correctly. While the CGI (and I use the term in only the broadest sense) birds are hilarious, my favorite scene is the \"retirement\" scene. I think that the word \"retirement\" is used six times in just three lines. Watch for it. It's hard to miss. BUT...it's strangely watchable. In contrast to many bad movies, it's not boring. Consider it a train wreck that you simply can't tear your eyes from. Even the boring scenes (and there really aren't that many) have some aspect (poor sound, hilariously unrealistic dialogue, odd things going on in the background) that fascinate. You'll be giggling and poking your elbow into the side of the person beside you. So, go rent it. Really. You won't regret it."}},{"title":"Manos: The Hands of Fate","releaseDate":"1966","rating":1.6,"direction":[{"director":"Harold P. Warren"}],"cast":[{"actor":"Tom Neyman"},{"actor":"John Reynolds"},{"actor":"Diane Adelson"},{"actor":"Harold P. Warren"},{"actor":"Stephanie Nielson"},{"actor":"Sherry Proctor"},{"actor":"Robin Redd"},{"actor":"Jackey Neyman Jones"},{"actor":"Bernie Rosenblum"},{"actor":"Joyce Molleur"},{"actor":"William Bryan Jennings"},{"actor":"Jay Hall"},{"actor":"Bettye Birns"},{"actor":"Lelanie Hansard"},{"actor":"Pat Coburn"},{"actor":"Pat Sullivan"},{"actor":"George Cavender"}],"comment":{"title":"Your eyes and ears will BLEED!","review":"Some films are so bad, they're good. Meaning \"fun\" or \"entertaining\" or at least \"interesting\"... But few films can claim to be so dreadful, you actually feel physical pain while viewing them. So bad are these damned few that you don't experience them or watch them: you \"endure\" them. This is the mother of all such films! \"Manos the hands of fate\" is without a doubt the most inept and atrociously awful film ever made. Its poorness is so extreme that of itself it is the film's strongest selling point. The script is non-existent, the acting makes Steven Seagal look like a member of the Royal Shakespeare Company and the editing could have been less horrendously botched by a blind Eskimo with no arms. It is also painfully slow. this film makes might barely last 70 minutes but you will feel like you've aged ten years by the end of it. That's what makes a film truly bad: the fact that despite its overbearing weaknesses it isn't even entertaining! Many people look back at the sixties and think, with obvious resentment for today's cinematic output, that \"they don't make them like this anymore!\". Watching \"Manos...\" would cure any breed of hardcore nostalgia. In the end I can not advise against this strongly enough. This is for the masochist in you (or the sadist if you insist on showing it to friends). Any other part of your person can only feel pained or offended by such extreme trash!"}},{"title":"Superbabies: Baby Geniuses 2","releaseDate":"2004","rating":1.5,"direction":[{"director":"Bob Clark"}],"cast":[{"actor":"Jon Voight"},{"actor":"Scott Baio"},{"actor":"Vanessa Angel"},{"actor":"Skyler Shaye"},{"actor":"Justin Chatwin"},{"actor":"Peter Wingfield"},{"actor":"Gerry Fitzgerald"},{"actor":"Leo Fitzgerald"},{"actor":"Myles Fitzgerald"},{"actor":"Max Iles"},{"actor":"Michael Iles"},{"actor":"Jared Scheideman"},{"actor":"Jordan Scheideman"},{"actor":"Maia Jae Bastidas"},{"actor":"Keana Lyn"},{"actor":"Joshua Lockhart"},{"actor":"Maxwell Lockhart"},{"actor":"Anastasia Trovato"}],"comment":{"title":"It is the best movie i have seen in yearsB","review":"The babys are not Even babyes they are 5 to 6 years old great acting the superbabyes is movie you should watch right now."}},{"title":"Kirk Cameron's Saving Christmas","releaseDate":"2014","rating":1.3,"direction":[{"director":"Darren Doane"}],"cast":[{"actor":"Kirk Cameron"},{"actor":"Darren Doane"},{"actor":"Bridgette Cameron"},{"actor":"Ben Kientz"},{"actor":"David Shannon"},{"actor":"Raphi Henley"},{"actor":"Cameron Ridenour"},{"actor":"Everett Ridenour"},{"actor":"Reese Ridenour"},{"actor":"Sharron Shannon"},{"actor":"Clarissa Shannon"},{"actor":"Lydia Shannon"},{"actor":"Andrew Shannon"},{"actor":"Jacob Wilson"},{"actor":"Jaidyn Wilson"},{"actor":"Jersi Wilson"},{"actor":"Anna Ybarra"},{"actor":"Jesse Brandt"}],"comment":{"title":"Every bit as bad as its reputation says","review":"This movie is so, so utterly wretched, not only in its execution but also in its conception. Just from a filmmaking perspective, it's extremely shoddy. Many portions look like home movies. A large part of the film is Cameron sitting in a car talking to a friend, and that's not interesting to watch. \"Comedy\" scenes have nothing to do with the rest of the story, and are creepily unfunny. A bizarre dance scene at the end is out of place and seems inserted to extend the time. Cameron's presence here is smarmy and smug; he lays out his spiritual views and his friend just gushes about how right Cameron is and how he's helped him see the light and love Christmas and all that. A bigger problem is the misinformation that Cameron spews so authoritatively, some of which runs counter to the Bible and to known history. To give a partial list: Cameron claims that Joseph and Mary were hiding from soldiers sent by Herod to kill all babies being born. But the Gospel of Matthew says the soldiers were sent by Herod AFTER Jesus was born, when Herod was tipped off by the visiting Wise Men that a King of the Jews had been born. Joseph and Mary then took Jesus and did the Flight into Egypt to save the child's life. Cameron insists the Nativity took place in a cave because the manger was made of stone. No source claims the manger was stone, or that the Nativity took place in a cave. Matthew says it was in Joseph's house in Bethlehem, Luke says it was a stable. This seems like a post-hoc attempt to link the Nativity with the Resurrection. Cameron says frankincense and myrrh were \"funeral spices.\" Although myrrh had been used in Egyptian mummification, frankincense was not, and both were much more commonly used as a sacred incense in Hebrew temples. He goes into detail about Saint Nicholas' life, presenting it as fact, but in truth very little is known for sure about him, and some Christian leaders openly suggest that he may never have existed. All that is known for sure is that someone named Nicholas was at one point the Bishop of Myra; everything else is unsure and in the territory of legend and myth. Legend has it that during the Council of Nicaea, Nicholas angrily struck Arius in the face for saying that Jesus and God were separate. Cameron depicts Nicholas savagely and brutally attacking Arius and beating him with a shepherd's crook, a scene many Christians found objectionable. Cameron depicts the historical Saint Nicholas climbing on a sleigh to deliver presents to children, something he never did in any legend. He also claims that Saint Nicholas is the gift-giver everywhere, also not supported by history. Some areas give gifts on Christmas, others at Epiphany, others on St. Nicholas' day, and in some areas the gifts are delivered by someone else entirely, such as La Befana in Italy, St. Basil in Greece, the Yule Lads in Iceland, St. Lucy in Croatia, and multiple countries have the gifts delivered by the Magi, angels, or the Christ Child himself. Cameron says Nicholas was \"sainted,\" when in reality Nicholas was never canonized. His sainthood was more by word-of-mouth. Cameron makes an elaborate rationalization for Christmas trees relating to the crucifixion and to the Garden of Eden; this connection is very convoluted and labored, and doesn't hold water. And he doesn't address the parts of Jeremiah which some feel are a commandment against Christmas trees. He claims that gifts piled under the Christmas tree are perfectly acceptable as they are reminiscent of the skyline of Jerusalem. By that reasoning, they could also be the skyline of Babylon or Rome, and could represent oppression and slavery. It's a poorly considered analogy that should never have been included. I'm sure I'm missing other bits, but I'm not inclined to go back and watch it again. It's unsettling to see Cameron stand and claim that it's OK to be materialistic at Christmas, because that's when God came to the Earth in material form. (That's pretty much a direct quote.) Never mind the many passages where the Bible tells us to set aside worldly things and not be materialistic! He exhorts viewers to eat themselves to bursting and buy the biggest ham and the richest butter....but isn't gluttony considered a deadly sin? At no point does he address such issues as helping the poor, feeding the hungry, giving to charity, volunteering, or anything. One is left with the impression that one should only think of one's self and one's immediate family and friends. Does Cameron remember Jesus' command to his followers to give away all they had to the poor? Somehow, I don't think so. Cameron gives the impression of someone who read parts of the Bible long ago, but rather than fit his life to the Bible, he is fitting the Bible to the life he wants to lead, and making one justification after another without ever bothering to double-check if he's remembering it correctly, or if there's something in the Bible that disagrees with him. It's clear he considers himself a better Christian than you. Many of the faithful have been turned off by this movie, some even going so far as to declare it blasphemous and call Cameron a false prophet. I'm not so sure I would agree, but at the same time, this movie does have some value of showing how even a faithful Christian can fall victim to the sins of pride and arrogance. It's clear that Cameron's ego was in overdrive, and this movie is not as much a testament to God as it is an expression of Cameron's arrogance."}}]